keps/sig-cluster-lifecycle/kubeadm/2568-kubeadm-non-root-control-plane/README.md

```diff
@@ -0,0 +1,1069 @@
+      allowPrivilegeEscalation: false
+      seccompProfile:
+        type: runtime/default
+      capabilities:
+        drop:
+        - ALL
+    image: k8s.gcr.io/kube-controller-manager:v1.21.0-beta.0.368_9850bf06b571d5-dirty
+    ... # omitted to save space.
+    volumeMounts:
+    - mountPath: /etc/kubernetes/pki
+      name: k8s-certs
+      readOnly: true
+    ... # omitted to save space. 
+```
+
+Here is a table of all the things that `kube-apiserver`, `kube-controller-manager`, `kube-scheduler` and `etcd` mount. 
+
+World readable stuff:
+- /usr/local/share/ca-certificates
+- /usr/share/ca-certificates
+- /etc/ssl/certs
+- /etc/ca-certificates
+
+<br />
+
+**Table key:-**
+
+| Abbreviation | Explanation |
+|--------------|-------------|
+| etcd | The UID/GID that we will assign to `etcd` |
+| kas  | The UID/GID that we will assign to `kube-apiserver` |
+| kcm | The UID/GID that we will assign to `kube-controller-manager` |
+| ks | The UID/GID that we will assign to `kube-scheduler` |
+| etcd_ca_crt_readers | The GID we will assign to a group that allows you to read /etc/kubernetes/pki/etcd/ca.crt |
+| ca_crt_readers | The GID we will assign to a group that allows you to read /etc/kubernetes/pki/etcd/ca.crt |
+| sa_key_readers | The GID we will assign to a group that allows you to read /etc/kubernetes/pki/sa.key |
+| front_proxy_ca_crt_readers | The GID we will assign to a group that allows you to read /etc/kubernetes/pki/front-proxy-ca.crt |
+
+<br/>
+
+**Stuff that we care about for this kep:-**
+| file/directory                                   | Components | File permission |
+| -------------------------------------------------|------------|-----------------|
+| /etc/kubernetes/pki/etcd/server.crt              | etcd       | 640 etcd etcd   |
+| /etc/kubernetes/pki/etcd/server.key              | etcd       | 640 etcd etcd   |
+| /etc/kubernetes/pki/etcd/peer.crt                | etcd       | 640 etcd etcd   |
+| /etc/kubernetes/pki/etcd/peer.key                | etcd       | 640 etcd etcd   |
+| /etc/kubernetes/pki/etcd/ca.crt                  | etcd, kas  | 640 root etcd_ca_crt_readers |
+| /var/lib/etcd/                                   | etcd       | 640 etcd etcd   |
+| /etc/kubernetes/pki/ca.crt                       | kas, kcm   | 640 root ca_crt_readers |
+| /etc/kubernetes/pki/apiserver-etcd-client.crt    | kas        | 640 kas kas     |
+| /etc/kubernetes/pki/apiserver-etcd-client.key    | kas        | 640 kas kas     |
+| /etc/kubernetes/pki/apiserver-kubelet-client.crt | kas        | 640 kas kas     |
+| /etc/kubernetes/pki/apiserver-kubelet-client.key | kas        | 640 kas kas     |
+| /etc/kubernetes/pki/front-proxy-client.crt       | kas        | 640 kas kas     |
+| /etc/kubernetes/pki/front-proxy-client.key       | kas        | 640 kas kas     |
+| /etc/kubernetes/pki/front-proxy-ca.crt           | kas, kcm   | 640 root front_proxy_ca_crt_readers |
```

Comment:
Pardon me if I missed this but could you clarify what `front_proxy_ca_crt_readers, ca_crt_readers, sa_key_readers `groups are ? Are these the shared groups between components?